core/builder.go

```diff
@@ -130,7 +130,16 @@ func setupNode(ctx context.Context, n *IpfsNode, cfg *BuildCfg) error {
 	}
 
 	var err error
-	n.Blockstore, err = bstore.WriteCached(bstore.NewBlockstore(n.Repo.Datastore()), kSizeBlockstoreWriteCache)
+	bs := bstore.NewBlockstore(n.Repo.Datastore())
+	n.Blockstore, err = bstore.WriteCached(bs, kSizeBlockstoreWriteCache)
+
+	rcfg, err := n.Repo.Config()
```

Comment:
I've screwed up here,  the `err != nil` check should be after 134.

Then latter I am checking for `err == nil` as we are requesting con-fig only in case of online more.